src/components/labware/PacbioRunWell.vue

```diff
@@ -196,17 +196,20 @@ const updateUsedAliquotSource = async (barcode) => {
   const tubeContent = store.tubeContentByBarcode(barcode)
   // Determine the source type for the used aliquot
   const source_type = tubeContent.type === 'pools' ? 'Pacbio::Pool' : 'Pacbio::Library'
+  const available_volume = store.getAvailableVolumeForAliquot({
+    sourceId: tubeContent.id,
+    sourceType: source_type,
+    volume: 0,
+  })
   /*id set to null because we are creating a new used aliquot. This is very important for calculating 
   the available volume where it checks for existing aliquots using id*/
-  const used_aliquot = createUsedAliquot(
-    { ...tubeContent, source_id: tubeContent.id, id: null },
-    () =>
-      store.getAvailableVolumeForAliquot({
-        sourceId: tubeContent.id,
-        sourceType: source_type,
-        volume: 0,
-      }),
-  )
+  const used_aliquot = createUsedAliquot({
+    ...tubeContent,
+    source_id: tubeContent.id,
+    id: null,
+    volume: available_volume,
+    available_volume,
+  })
```

Comment:
Just a thought: now that we are dynamically calculating available_volume every time, would it be better to refactor the createUsedAliquot function to handle the calculation of available_volume internally so we don't need to repeat two calls? This would make the code cleaner.

Reply:
available_volume requires the run store state so we need to pass the whole state into the createUsedAliquot function. I am also not sure available_volume should live in the aliquot as its actually a mixture of the aliquot data and library/pool data but couldn't find a better place to put it.

Reply:
Ok, hmm, that makes sense. yes, it looks like getAvailableVolumeForAliquot needs lots of store state variables.